tnt/src/main/java/org/dcsa/conformance/standards/tnt/checks/TntChecks.java

```diff
@@ -36,75 +35,111 @@ public class TntChecks {
-
     var checks = new ArrayList<JsonContentCheck>();
 
+    checks.add(
+        JsonAttribute.customValidator(
+            "Validate response has events",
+            body -> {
+              Set<String> issues = new LinkedHashSet<>();
+              ArrayList<JsonNode> eventNodes = TntHelper.findEventNodes(body, issues);
+              if (eventNodes.isEmpty()) {
+                issues.add("No events found in response");
+              }
+              return issues;
+            }));
+
     // The OpenAPI specification explicitly states that the eventIDs must be unique.
     checks.add(
         JsonAttribute.customValidator(
             "Validate that eventID values are unique",
-            body ->
-                StreamSupport.stream(body.spliterator(), false)
-                        .filter(node -> node.has("eventID"))
-                        .map(node -> node.path("eventID").asText())
-                        .allMatch(new HashSet<>()::add)
-                    ? Set.of()
-                    : Set.of("Event ID values are not unique")));
+            body -> {
+              Set<String> issues = new HashSet<>();
+              ArrayList<JsonNode> eventNodes = TntHelper.findEventNodes(body, issues);
+              Set<String> result =
+                  eventNodes.stream()
+                          .filter(node -> node.has("eventID"))
+                          .map(node -> node.path("eventID").asText())
+                          .allMatch(new HashSet<>()::add)
+                      ? Set.of()
+                      : Set.of("Event ID values are not unique");
+              issues.addAll(result);
+              return issues;
+            }));
 
     checks.add(
         JsonAttribute.customValidator(
             "Validate eventCreatedDateTime parameter conditions are met",
             body -> {
+              Set<String> issues = new LinkedHashSet<>();
+              ArrayList<JsonNode> eventNodes = TntHelper.findEventNodes(body, issues);
+
               Set<OffsetDateTime> eventCreatedDateTimes =
-                  StreamSupport.stream(body.spliterator(), false)
-                      .map(node -> OffsetDateTime.parse(node.path("eventCreatedDateTime").asText()))
+                  eventNodes.stream()
+                      .flatMap(
+                          node -> {
+                            try {
+                              return Stream.of(
+                                  OffsetDateTime.parse(node.path("eventCreatedDateTime").asText()));
+                            } catch (DateTimeParseException e) {
+                              issues.add(
+                                  "Invalid eventCreatedDateTime format: "
+                                      + node.path("eventCreatedDateTime").asText()
+                                      + ": "
+                                      + e.getMessage());
+                              return Stream.empty(); // Return empty stream for invalid date-times
```

Comment:
I thought returning an empty stream from the Catch block, seems to be weird .hence added the comment. Anyhow this will go away with the improvised logic.